### ISSUE TITLE
Overgrown php-error.log results in 504 on wp-admin because of Seravo Plugin

### ISSUE THREAD

**Issue by nuuttitoivola** [Seravo/seravo-plugin#416]
**Describe the bug**
If a site has a `php-error.log` that has grown huge(40-100GB+ in filesize) for some reason, then the `example.com/wp-admin` will result to a 504 in the client browser. The issue seems to be that the Seravo Plugin tries to read the log file when entering  the dashboard-page.

**To Reproduce**
1. Fill up your `php-error.log` with PHP notices/warngings/errors so that it's (maybe)at least 40GB in size
2. Go to `example.com/wp-admin` & try to log in if not already 
3. Wait a couple of minutes
4. Experience the status code 504

**Expected behavior**
The path `example.com/wp-admin` should lead to the WordPress admin panel without issues.

**Desktop (please complete the following information):**
 - OS: Linux(Fedora) & Windows 10
 - Browser Chrome & Firefox

**Additional context**
- The log in was registered as SUCCESS in `wp-login.log` and after renaming the `php-error.log` the refresh of the `/wp-admin` page resulted in working admin. 
- Seravo Plugin version was 1.9.18 on the problematic site.
- It's unclear what size the `php-error.log` should be for the error to occur.

**Comment by ottok** [comment]
To read about recent errors we use `read_log_lines_backwards` to parse just the last lines of the log, see https://github.com/Seravo/seravo-plugin/blob/master/modules/login-notification.php#L93 and there is a max size argument, but maybe it does not work?

Perhaps this is needed?
```
$output = Logs::read_log_lines_backwards('/data/log/wp-login.log', -1, self::$max_rows);
-->
$output = Logs::read_log_lines_backwards('/data/log/wp-login.log', -1, null, self::$max_rows);
```

https://github.com/Seravo/seravo-plugin/blob/940d2e4ec4c0e62ec9d791e2586060f00996abcb/modules/logs.php#L318
```
public static function read_log_lines_backwards( $filepath, $offset = -1, $lines = 1, $regex = null, $cutoff_bytes = null ) {
```

While at it, grep for all occurences of `read_log_lines_backwards` and ensure it is used correctly.

**Comment by Wiljami** [comment]
In 7944fcb01598b5502e332e760c73920545b0e379 read_log_lines_backward was made to abort the reading php-error.log if it's too large.

The admin panel should be accessible in the latest seravo-plugin versions.

**Comment by ottok** [comment]
The file size check only masks the problem. The file size should not effect
reading lines from the end of a file at all. "tail x.log" takes equal time
for a 1 MB file as for a 1000 MB file.

**Comment by Moppa5** [comment]
It seems that the count of words and characters per row are also an issue. For instance a few million characters on a row caused the php-error.log box showing only "Critical error on your website --" and taking a long time to load the logs page in general.

**Comment by Wiljami** [comment]
> The file size check only masks the problem. The file size should not effect reading lines from the end of a file at all. "tail x.log" takes equal time for a 1 MB file as for a 1000 MB file.

Fix in #421 indeed works with small and massive (10GB) log files.

**Comment by Wiljami** [comment]
Since there has been no reports of this issue repeating after the fix made last August, I believe this is fixed.

This should be reopened if it occurs again.
